go/adbc/driver/snowflake/driver.go

```diff
@@ -228,17 +229,24 @@ func (d *driverImpl) NewDatabase(opts map[string]string) (adbc.Database, error)
 }
 
 func (d *driverImpl) NewDatabaseWithContext(ctx context.Context, opts map[string]string) (adbc.Database, error) {
-	return d.NewDatabaseWithOptions(ctx, opts)
+	return d.NewDatabaseWithOptionsContext(ctx, opts)
 }
 
 func (d *driverImpl) NewDatabaseWithOptions(
+	opts map[string]string,
+	optFuncs ...Option,
+) (adbc.Database, error) {
+	return d.NewDatabaseWithOptionsContext(context.Background(), opts, optFuncs...)
+}
+
+func (d *driverImpl) NewDatabaseWithOptionsContext(
 	ctx context.Context,
 	opts map[string]string,
 	optFuncs ...Option,
 ) (adbc.Database, error) {
 	opts = maps.Clone(opts)
 
-	dbBase, err := driverbase.NewDatabaseImplBase(ctx, &d.DriverImplBase)
+	dbBase, err := driverbase.NewDatabaseImplBase(context.Background(), &d.DriverImplBase)
```

Comment:
why the switch back to `context.Background()` if we're taking the `ctx` in as an argument?

Reply:
Sorry. missed that.